src/botPage/view/View.js

```diff
@@ -627,31 +635,54 @@ export default class View {
         this.blockly.stop();
     }
     addEventHandlers() {
+        const getRunButtonElements = () => document.querySelectorAll('#runButton, #summaryRunButton');
+        const getStopButtonElements = () => document.querySelectorAll('#stopButton, #summaryStopButton');
+
         window.addEventListener('storage', e => {
             window.onbeforeunload = null;
             if (e.key === 'activeToken' && !e.newValue) window.location.reload();
             if (e.key === 'realityCheckTime') hideRealityCheck();
         });
 
         globalObserver.register('Error', error => {
-            $('#runButton, #summaryRunButton').prop('disabled', false);
+            getRunButtonElements().forEach(el => {
+                const elRunButton = el;
+                elRunButton.removeAttribute('disabled');
+            });
+
             if (error.error && error.error.error.code === 'InvalidToken') {
                 removeAllTokens();
                 updateTokenList();
                 this.stop();
             }
         });
 
+        globalObserver.register('bot.running', () => {
```

Comment:
@aaron-binary does this `globalObserver` have something like `reaction` so it listen to state changes and when ever bot state change to running or not running , disable and enable the `buttons` once only.

Reply:
Through the observer you can `observer.register('name')` to a named event, once that event is emitted (through `observer.emit('name')`) the callback will be executed. In some cases, the buttons should not be enabled when the bot is not running, e.g. when the user is fetching `contracts_for` from the API.